tests/test_errors.py

```diff
@@ -517,6 +521,137 @@ def test_all_errors():
         pytest.fail('core_schema.ErrorType needs to be updated')
 
 
+def test_validation_error_cause_usage():
+    if sys.version_info < (3, 11):
+        from exceptiongroup import BaseExceptionGroup
+    else:
+        from builtins import BaseExceptionGroup
+
+    enabled_config: CoreConfig = {'validation_error_cause': True}
+
+    def check_grouped_exception(exc: BaseException) -> BaseException:
+        """
+        Handled the difference pre 3.11 and after,
+        returns the exception to keep testing with:
+        pre 3.11 and extra UserWarning exception gets added, so unwrapping that.
+        """
+        if sys.version_info < (3, 11):
+            assert repr(exc).startswith("UserWarning('Pydantic: ")
+            assert exc.__cause__ is not None
+            return exc.__cause__
+        else:
+            assert exc.__notes__
+            assert exc.__notes__[-1].startswith('\nPydantic: ')
+            return exc
+
+    # Important to test a singular as well as a multi,
+    # because the singular originally had problems with losing the traceback
+    def singular_raise_py_error(v: Any) -> Any:
+        raise ValueError('Oh no!')
+
+    s1 = SchemaValidator(core_schema.no_info_plain_validator_function(singular_raise_py_error), config=enabled_config)
+    with pytest.raises(ValidationError) as exc_info:
+        s1.validate_python('anything')
+
+    assert isinstance(exc_info.value.__cause__, BaseExceptionGroup)
+    assert len(exc_info.value.__cause__.exceptions) == 1
+    cause = exc_info.value.__cause__.exceptions[0]
+    cause = check_grouped_exception(cause)
+    assert repr(cause) == repr(ValueError('Oh no!'))
+    assert cause.__traceback__ is not None
+
+    def multi_raise_py_error(v: Any) -> Any:
+        try:
+            raise AssertionError('Wrong')
+        except AssertionError as e:
+            raise ValueError('Oh no!') from e
+
+    s2 = SchemaValidator(core_schema.no_info_plain_validator_function(multi_raise_py_error), config=enabled_config)
+    with pytest.raises(ValidationError) as exc_info:
+        s2.validate_python('anything')
+
+    def validate_s2_chain(val_err: ValidationError):
```

Comment:
better not to use functions like this as it becomes hard to understand upon failure, especially as it only removes one duplication.